ui/e2e/003-post-note-with-markdown.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { test, expect } from '@playwright/test';
+import shipManifest from './shipManifest.json';
+
+// mardev is the owner of the group
+// patbud is the testing ship
+
+const testUrl = `${shipManifest['~habduc-patbud'].webUrl}/apps/groups/`;
+const groupOwner = 'mardev';
+
+// Authenticate as patbud
+test.use({ storageState: 'e2e/.auth/habduc-patbud.json' });
+
+test('Add markdown note to existing notebook channel', async ({ page }) => {
+  test.skip(process.env.APP === 'chat', 'skip on talk');
+  await page.goto(testUrl);
+  await page.getByRole('link', { name: `${groupOwner} test group` }).waitFor();
+  await page.getByRole('link', { name: `${groupOwner} test group` }).click();
+  await page.getByTestId(`sidebar-item-text-${groupOwner} notebook`).waitFor();
+  await page.getByTestId(`sidebar-item-text-${groupOwner} notebook`).click();
+  await page.getByTestId('add-note-button').waitFor();
+  await page.getByTestId('add-note-button').click();
+  await page.getByTestId('note-title-input').waitFor();
+  await page.getByTestId('note-title-input').click();
+  await page.getByTestId('note-title-input').fill('Markdown note');
+
+  // Markdown option by default should be false
+  await expect(
+    page.getByTestId('edit-with-markdown-toggle').getByRole('button')
+  ).toHaveAttribute('aria-pressed', 'false');
+
+  await page
+    .getByTestId('edit-with-markdown-toggle')
+    .getByRole('button')
+    .click();
+  await expect(
+    page.getByTestId('edit-with-markdown-toggle').getByRole('button')
+  ).toHaveAttribute('aria-pressed', 'true');
+
+  await page.getByTestId('note-markdown-editor').click();
+  await page.getByTestId('note-markdown-editor').fill('## Heading level 2');
+  await page.getByTestId('save-note-button').click();
```

Comment:
Do we actually need to use test ids or can we target these with text? Wondering if we have an accessibility issue?

Reply:
It's just for practicality.
Using test IDs is a decision I made (maybe should have been a recommendation to discuss).

In my experience, they make e2e tests easier to maintain and prevent false negatives.
If we want to diagnose accessibility issues we can use any tool specifically designed for that, and will get better coverage and realistic insights.

My preference for using test IDs is based on making explicit when an element is targeted by a functional test.
Otherwise, when changes are made to behaviors or text based on business or design decisions, tests may break unless their locators are updated accordingly.
If it's not clear at a glance that a component I'm modifying is expected (by the tests) to be there, do something, or read something, I may make my changes and omit updating the tests. Meaning I'll have to come back once they were merged and the failing tests triggered.

If we prefer to use text, I can make the changes. LMK

Reply:
yeah that makes sense to me